### ISSUE TITLE
Send binary data to client

### ISSUE THREAD

**Issue by leon** [EpicGames/PixelStreamingInfrastructure#297]
## Component your question relates to
- [ ] Signalling server
- [x] Frontend library
- [ ] Frontend UI library
- [ ] Matchmaker
- [ ] SFU
- [ ] This repo
- [ ] Other

## Background
I'm programmatically calling `HighResShot` and reading back the generated screenshot.
I then want to send this to the browser so the user can download the image.

## Question
I tried base64 encoding the image and sending it in a normal message, but that only works up to a certain resolution, then it crashes.

How can I send binary data back to the browser?

**Comment by mcottontensor** [comment]
We send screenshots back to the browser when we do freeze frames by sending the data over the data channel. Is this what you're doing? I'm curious why it would be crashing. Do you have a log or stack trace of the log?

You can check FreezeFrame.cpp for references to how we do it.

**Comment by lukehb** [comment]
In addition to Matt's answer, as he said, if you study FreezeFrame.cpp you will see we end up chunking the data into multiple messages if the payload is beyond a certain size (I think from memory webrtc has like 16mb limit for a single data channel message).

tldr; Maybe try chunking your payload?

**Comment by leon** [comment]
Thanks for the answers
We had to find a workaround so we ended up sending a s3 signed url from the frontend and have unreal upload to the signed url and then send the url back.

We might revisit this later on and se if maybe pixel streaming 2 works better sending user generated binary data. (for us)

but I will close this for now
